internal/tiltfile/tiltfile_state.go

```diff
@@ -685,25 +696,28 @@ func (s *tiltfileState) assembleImages() error {
 }
 
 func (s *tiltfileState) assembleDC() error {
-	if len(s.dc.services) > 0 && !container.IsEmptyRegistry(s.defaultReg) {
+	if s.dc.ServiceCount() > 0 && !container.IsEmptyRegistry(s.defaultReg) {
 		return errors.New("default_registry is not supported with docker compose")
 	}
 
-	for _, svc := range s.dc.services {
-		builder := s.buildIndex.findBuilderForConsumedImage(svc.ImageRef())
-		if builder != nil {
-			// there's a Tilt-managed builder (e.g. docker_build or custom_build) for this image reference, so use that
-			svc.ImageMapDeps = append(svc.ImageMapDeps, builder.ImageMapName())
-		} else {
-			// create a DockerComposeBuild image target and consume it if this service has a build section in YAML
-			err := s.maybeAddDockerComposeImageBuilder(svc)
-			if err != nil {
-				return err
+	for _, resSet := range s.dc {
+		for _, svcName := range resSet.serviceNames {
+			svc := resSet.services[svcName]
+			builder := s.buildIndex.findBuilderForConsumedImage(svc.ImageRef())
+			if builder != nil {
+				// there's a Tilt-managed builder (e.g. docker_build or custom_build) for this image reference, so use that
+				svc.ImageMapDeps = append(svc.ImageMapDeps, builder.ImageMapName())
+			} else {
+				// create a DockerComposeBuild image target and consume it if this service has a build section in YAML
+				err := s.maybeAddDockerComposeImageBuilder(svc)
+				if err != nil {
+					return err
+				}
 			}
+			// TODO(maia): throw warning if
+			//  a. there is an img ref from config, and img ref from user doesn't match
+			//  b. there is no img ref from config, and img ref from user is not of form .*_<svc_name>
```

Comment:
Not sure if these todos are valid anymore? Delete?

Reply:
:+1: